brightness-mode-explainer.md

```diff
@@ -0,0 +1,129 @@
+- [make up mirror](https://play.google.com/store/apps/details?id=mmapps.mirror.pro&hl=en&gl=US) style apps
+
+[Screen Wake Lock API issue 129](https://github.com/w3c/screen-wake-lock/issues/129) was filed back in 2018, and since then there has been feedback from developers working on multiple areas about the usefulness of an API that addresses this use case.
+
+## Open design issues
+
+- Things to consider in the spec proposal
+  - UAs should be free to limit maximum brightness level and/or do other things like increase contrast.
+  - Idea is to focus on mobile first (i.e. not worry about external displays). How to indicate that in the spec? Does it *need* to be indicated in the spec?
+  - Should this trigger "`prefers-contrast: more`" in [CSS Media Queries](https://drafts.csswg.org/mediaqueries-5/#prefers-contrast)?
+- From [https://github.com/w3c/devicesensors-wg/issues/51](https://github.com/w3c/devicesensors-wg/issues/51)
+  - Brightness levels
+    - How bright is too bright? Consider 100% brightness with HDR displays, for example.
+    - Take a granular value or an enum?
+    - Related to whether script should be allowed to reduce brightness.
+  - Permission model
+    - Probably require user gesture (request but not consume it).
+    - How to deal with multiple or external displays
+  - While brightness changes
+    - What if users change the brightness level while the lock is held?
+  - When dropping screen brightness request
+      - Does the UA have to restore the previous brightness level?
+        - What about external displays? Do UAs need to keep track of levels for each display?
+      - Should script be allowed to "hold the lock" indefinitely?
+
+## Goals
+
+- Provide the ability to request an increase to the display's brightness. This could be a UA-controlled maximum (which *could* be the maximum display brightness), either indefinitely or for a period of time.
+- Provide the ability to release the request so that the device's brightness returns to its pre-request value (i.e. hand back control to OS).
+- Handle error cases, where such requests are denied or not possible.
+
+## Non-goals
+
+- Reading screen brightness level.
+- Adjusting display brightness level to arbitrary values (absolute or relative).
+  - [\<video\> integration to allow e.g. granular brightness control](https://github.com/w3c/screen-wake-lock/issues/129#issuecomment-926603108) is an instance of the above. This is most likely better handled elsewhere and should probably be a UA-specific control.
+
+## API
+### Web IDL
+
+Extend the [Screen interface](https://drafts.csswg.org/cssom-view/#the-screen-interface) with a new operation, inspired by the [Wake Lock API](https://developer.mozilla.org/en-US/docs/Web/API/WakeLockSentinel). Something like:
+
+```webidl
+[SecureContext]
+partial interface Screen {
+  Promise<ScreenBrightnessSentinel> requestBrightnessIncrease();
+};
+
+interface ScreenBrightnessSentinel {
+  Promise<undefined> release();
+};
+```
+
```

Comment:
I respectfully object to this being added here as we never settled on an API (i.e., we wouldn't need this document if we had settled on a particular API already). 

I asked that this be moved to the "Proposed Solutions" section, along with the other proposed solutions. 

This should be moved to the the proposed solutions section.

Reply:
Done. I honestly thought we'd settled on this being the current proposal (which is why I'd originally listed the other ideas as "considered alternatives" rather than "proposed solutions" in the document), but this makes it easier to understand why we have "proposed solutions" in the first place.